src/main/java/controller/UserAPI.java

```diff
@@ -100,7 +100,8 @@
         requestValidator.loginRequestValidator(data.keySet(), path);
         String username = data.get("Username").toString();
         String password = data.get("Password").toString();
-        long id = userAccessor.login(username, password);
+        String passwordHash = User.hashPassword(password);
+        long id = userAccessor.login(username, passwordHash);
         if(id == -1) {
```

Comment:
Did we make sure, that the random generated id can never be negative (or -1 to be specific)?

Reply:
Yes the range of random numbers being generated is from 1 to `Long.MAX_VALUE`.